src/modalForms/AddDepartmentForm.tsx

```diff
@@ -0,0 +1,93 @@
+import { Formik } from 'formik';
+import { FC } from 'react';
+import { useDispatch } from 'react-redux';
+import { fetchNewDepartment } from '../redux/departments/actions';
+import { IDepartment } from '../types/departments';
+import * as yup from 'yup';
+
+const AddDepartmentForm: FC = () => {
+  const dispatch = useDispatch();
+  const validationsSchema = yup.object().shape({
+    name: yup
+      .string()
+      .required()
+      .min(2, 'Name is too short - should be 2 chars minimum.')
+      .max(10, 'Name is too long - should be 10 chars maximum.'),
+    description: yup
+      .string()
+      .required()
+      .min(5, 'Description is too short - should be 5 chars minimum.')
+      .max(50, 'Description is too long - should be 50 chars maximum.'),
+  });
+  
+  function createNewDepartment(values: IDepartment) {
+    const newDepartment: IDepartment = {
+      name: values.name,
+      description: values.description,
+    };
+    dispatch(fetchNewDepartment(newDepartment));
+  }
+
+  return (
+      <Formik
+        initialValues={{ name: '', description: '' }}
+        validateOnBlur
+        onSubmit={(values) => createNewDepartment(values)}
```

Comment:
try avoiding the arrow functions directly in body. Use `useCallback` for such purposes